x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/JdbcDateUtils.java

```diff
@@ -47,7 +51,9 @@ static long asMillisSinceEpoch(String date) {
     }
     
     static Date asDate(String date) {
-        return new Date(utcMillisRemoveTime(asMillisSinceEpoch(date)));
+        ZonedDateTime zdt = ISO_WITH_MILLIS.parse(date, ZonedDateTime::from);
+        ZoneId zoneId = zdt.getZone();
+        return new Date(zdt.toLocalDate().atStartOfDay(zoneId).withZoneSameInstant(zoneId).toInstant().toEpochMilli());
```

Comment:
It's replaced with `atStartOfDay` with the correct timezone vs the calculation that there was there before that was done on UTC millis so it was setting the start of date in UTC instead of the actual tz (`2019-03-12 00:00:00+00:00` instead for example `2019-03-13 00:00:00+10:00`)